ipa/scripts/cli.py

```diff
@@ -286,59 +292,124 @@ def test(access_key_id,
         sys.exit(1)
 
 
-@click.command()
-@click.option(
-    '--clear',
-    is_flag=True,
-    help='Clear list of results history.'
-)
+@click.group(invoke_without_command=True)
 @click.option(
     '--history-log',
     default=IPA_HISTORY_FILE,
     type=click.Path(exists=True),
     help='Location of the history log file to display results from.'
 )
-@click.option(
-    '--list',
-    'list_results',
-    is_flag=True,
-    help='Display list of results history.'
+@click.pass_context
+def results(context, history_log):
+    """Process provided history log and results files."""
+    if context.obj is None:
+        context.obj = {}
+    context.obj['history_log'] = history_log
+
+    if context.invoked_subcommand is None:
+        context.invoke(show, item=1)
+
+
+@click.command()
+@click.pass_context
+def clear(context):
+    """
+    Clear the results from the history file.
+    """
+    ipa_utils.update_history_log(context.obj['history_log'], clear=True)
+
+
+@click.command()
+@click.argument(
+    'item',
+    type=click.INT
 )
+@click.pass_context
+def delete(context, item):
+    """
+    Delete the specified history item from the history log.
+    """
+    history_log = context.obj['history_log']
+    no_color = context.obj['no_color']
+    try:
+        with open(history_log, 'r+') as f:
+            lines = f.readlines()
+            history = lines.pop(len(lines) - item)
+            f.seek(0)
+            f.write(''.join(lines))
```

Comment:
Should flush the file before calling truncate. The Python documentation is not explicit about the flush, it is pointed out in the Python doc then if one calls truncate from the os module flush and sync are required ahead of time. Since we do not know the internals lets at least flush the content of the file out of the python buffers before truncating the file size.

Reply:
I don't think calling flush before truncate does anything. But it doesn't hurt anything either. But looking at the cpython unit tests there's no use of explicit flush. https://github.com/python/cpython/blob/a5af6e1af77ee0f9294c5776478a9c24d9fbab94/Lib/test/test_fileio.py#L487

On close the file is flushed.